esmvalcore/_recipe/recipe.py

```diff
@@ -1087,179 +878,33 @@ def _extract_preprocessor_order(profile):
-                _add_extra_facets(var, config_user['extra_facets_dir'])
-                _update_timerange(var, config_user)
-                files = _get_input_files(var, config_user)[0]
-                if var.get('optional') and not files:
-                    logger.info(
-                        "Skipping: no data found for %s which is marked as "
-                        "'optional'", var)
-                else:
-                    append(group_prefix, var)
-                    timeranges.add(var['timerange'])
-            _check_differing_timeranges(timeranges, required_vars)
-            variable['timerange'] = " ".join(timeranges)
-
-    # An empty derive_input (due to all variables marked as 'optional' is
-    # handled at a later step
-    return derive_input
-
-
-def _get_preprocessor_task(variables, profiles, config_user, task_name):
-    """Create preprocessor task(s) for a set of datasets."""
-    # First set up the preprocessor profile
-    variable = variables[0]
-    preproc_name = variable.get('preprocessor')
-    if preproc_name not in profiles:
-        raise RecipeError(
-            "Unknown preprocessor {} in variable {} of diagnostic {}".format(
-                preproc_name, variable['short_name'], variable['diagnostic']))
-    profile = deepcopy(profiles[variable['preprocessor']])
-    logger.info("Creating preprocessor '%s' task for variable '%s'",
-                variable['preprocessor'], variable['short_name'])
-    variables = _limit_datasets(variables, profile,
-                                config_user.get('max_datasets'))
-    for variable in variables:
-        _add_cmor_info(variable)
-    # Create preprocessor task(s)
-    derive_tasks = []
-    # set up tasks
-    if variable.get('derive'):
-        # Create tasks to prepare the input data for the derive step
-        derive_profile, profile = _split_derive_profile(profile)
-        derive_input = _get_derive_input_variables(variables, config_user)
-
-        for derive_variables in derive_input.values():
-            for derive_variable in derive_variables:
-                _add_cmor_info(derive_variable, override=True)
-            derive_name = task_name.split(
-                TASKSEP)[0] + TASKSEP + derive_variables[0]['variable_group']
-            task = _get_single_preprocessor_task(
-                derive_variables,
-                derive_profile,
-                config_user,
-                name=derive_name,
-            )
-            derive_tasks.append(task)
-
-    # Create (final) preprocessor task
-    task = _get_single_preprocessor_task(
-        variables,
-        profile,
-        config_user,
-        ancestor_tasks=derive_tasks,
-        name=task_name,
-    )
-
-    return task
-
-
 class Recipe:
     """Recipe object."""
 
-    info_keys = ('project', 'activity', 'driver', 'dataset', 'exp',
-                 'sub_experiment', 'ensemble', 'version')
+    info_keys = ('project', 'activity', 'dataset', 'exp', 'ensemble',
```

Comment:
Is this needed? It looks like the alias generation is using  _ALIAS_INFO_KEYS in `to_datasets.py`

Reply:
Well spotted, I removed it in https://github.com/ESMValGroup/ESMValCore/pull/1609/commits/dec030636353e98abdfa4fdffcf85521d4eb7a7b.